### ISSUE TITLE
stddef.h not found

### ISSUE THREAD

**Issue by feux-follets** [include-what-you-use/include-what-you-use#1262]
System version:
Arch WSL
clang 15
gcc 13
iwyu compiled with clang 15

```c++
#include <string>
int main(){
        return 0;
}
```

```
$ include-what-you-use main.cpp
In file included from main.cpp:1:
In file included from /usr/lib64/gcc/x86_64-pc-linux-gnu/13.1.1/../../../../include/c++/13.1.1/string:42:
In file included from /usr/lib64/gcc/x86_64-pc-linux-gnu/13.1.1/../../../../include/c++/13.1.1/bits/char_traits.h:42:
In file included from /usr/lib64/gcc/x86_64-pc-linux-gnu/13.1.1/../../../../include/c++/13.1.1/bits/postypes.h:40:
In file included from /usr/lib64/gcc/x86_64-pc-linux-gnu/13.1.1/../../../../include/c++/13.1.1/cwchar:44:
/usr/include/wchar.h:35:10: fatal error: 'stddef.h' file not found
#include <stddef.h>
         ^~~~~~~~~~
```

**Comment by kimgr** [comment]
Did you build iwyu yourself using Cmake? If so, the cmake steps should have copied the clang builtin headers into the right place in the build dir. Can you say a bit more about how you built?

**Comment by feux-follets** [comment]
> Did you build iwyu yourself using Cmake? If so, the cmake steps should have copied the clang builtin headers into the right place in the build dir. Can you say a bit more about how you built?

Checked out clang_15 branch.

```
cmake -DCMAKE_PREFIX_PATH=/usr/lib ../include-what-you-use
```

LLVM libs are located under /usr/lib
```
-rwxr-xr-x  1 root root  102K Apr  8 13:45 LLVMgold.so*
lrwxrwxrwx  1 root root    13 Apr  8 13:45 libLLVM.so -> libLLVM-15.so*
-rw-r--r--  1 root root  755K Apr  8 13:45 libLLVMDemangle.a
-rw-r--r--  1 root root  5.9M Apr  8 13:45 libLLVMSupport.a
-rw-r--r--  1 root root  1.2M Apr  8 13:45 libLLVMTableGen.a
-rwxr-xr-x  1 root root  121M Apr  8 13:45 libLLVM-15.so*
lrwxrwxrwx  1 root root    13 Apr  8 13:45 libLLVM-15.0.7.so -> libLLVM-15.so*
```

I can see the stddef.h inside build directory:

```
[evan@Evan-Desktop include]$ readlink -f stddef.h
/home/evan/include-what-you-use/build/lib/clang/15.0.7/include/stddef.h
```

**Comment by kimgr** [comment]
Are you also running `include-what-you-use` from the build directory?

If not, we haven't yet figured out how to install IWYU in a sane way. See https://github.com/include-what-you-use/include-what-you-use/blob/master/README.md#how-to-install for details on how to wire it.

**Comment by feux-follets** [comment]
> Are you also running `include-what-you-use` from the build directory?
> 
> If not, we haven't yet figured out how to install IWYU in a sane way. See https://github.com/include-what-you-use/include-what-you-use/blob/master/README.md#how-to-install for details on how to wire it.

I have installed it to another directory and added the directory to PATH.

**Comment by kimgr** [comment]
OK, in that case you need to make the Clang builtin headers available some other way (easiest is probably to install clang). I'll close this as a duplicate of #100.

**Comment by Krzmbrzl** [comment]
Not related to the actually reported issue here, but maybe relevant for folks landing here via Google:
I had `include-what-you-use` installed via my distribution's (Ubuntu 22.04) package manager and I also had `clang` installed. However, the version of clang that `iwyu` was built against was different than the default `clang` version that I had installed.
I had to check
```
include-what-you-use --version
```
which yielded
```
include-what-you-use 0.17 based on Ubuntu clang version 13.0.1-2ubuntu2.2
```
and thus I needed to `sudo apt install clang-13` specifically (I had `clang-14` installed) in order to make `iwyu` find the respective include (`stddef.h`).
That is to say: The version of clang installed must match the version of clang that iwyu is built against.
